arangod/Aql/AqlTransaction.cpp

```diff
@@ -74,8 +79,12 @@ AqlTransaction::AqlTransaction(
 
     return true;
   });
+  
+  TRI_ASSERT(state() != nullptr);
 }
 
+AqlTransaction::~AqlTransaction() = default;
```

Comment:
It is quite common in the arango codebase to explicitly define the destructor (even if it is just the default implementation). I am actually not a fan of this, not just because an explicit definition makes the destructor user defined (even if it is just the default implementation), which potentially prevents the compiler from performing some optimizations. So I am fine with removing it! :)